From: Dominik George <natureshadow@debian.org>
Subject: Fix validation of http proxy setting
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=920401
Bug: https://github.com/midori-browser/core/issues/243
--- a/core/preferences.vala
+++ b/core/preferences.vala
@@ -166,6 +166,10 @@ namespace Midori {
                 }
             }
             proxy.search_changed.connect (() => {
+                if (! ("://" in settings.http_proxy)) {
+                    proxy.get_style_context ().add_class ("error");
+                    return;
+                }
                 string[] parts = settings.http_proxy.split ("://", 2);
                 if (parts[1] == "" || ":" in parts[1] || "/" in parts[1]) {
                     proxy.get_style_context ().add_class ("error");
